lib/experimental/fonts-api/class-wp-fonts-resolver.php

```diff
@@ -191,37 +191,38 @@ private static function get_settings() {
 
 		foreach ( $variations as $variation ) {
 
-			// Skip if settings.typography.fontFamilies are not defined in the variation.
-			if ( empty( $variation['settings']['typography']['fontFamilies'] ) ) {
+			// Skip if settings.typography.fontFamilies.theme is not defined or is not an array.
+			if (
+				! isset( $variation['settings']['typography']['fontFamilies']['theme'] ) ||
+				! is_array( $variation['settings']['typography']['fontFamilies']['theme'] )
+			) {
 				continue;
 			}
 
 			// One time, set any missing parts of the array structure.
 			if ( $set_theme_structure ) {
 				$set_theme_structure = false;
 				$settings            = static::set_tyopgraphy_settings_array_structure( $settings );
-			}
 
-			// Initialize the font families from variation if set and is an array, otherwise default to an empty array.
-			$variation_font_families = ( isset( $variation['settings']['typography']['fontFamilies']['theme'] ) && is_array( $variation['settings']['typography']['fontFamilies']['theme'] ) )
-				? $variation['settings']['typography']['fontFamilies']['theme']
-				: array();
+				// Font families from settings must be an array.
+				if (
+					! isset( $settings['typography']['fontFamilies']['theme'] ) ||
+					! is_array( $settings['typography']['fontFamilies']['theme'] )
+				) {
+					$settings['typography']['fontFamilies']['theme'] = array();
+				}
+			}
 
-			// Merge the variation settings with the global settings.
+			// Merge the variation settings with the global settings. Variations overwrite the globals.
 			$settings['typography']['fontFamilies']['theme'] = array_merge(
 				$settings['typography']['fontFamilies']['theme'],
-				$variation_font_families
+				$variation['settings']['typography']['fontFamilies']['theme']
 			);
-
-			// Make sure there are no duplicates.
-			$settings['typography']['fontFamilies'] = array_unique( $settings['typography']['fontFamilies'], SORT_REGULAR );
-
-			// The font families from settings might become null after running the `array_unique`.
-			if ( ! isset( $settings['typography']['fontFamilies']['theme'] ) || ! is_array( $settings['typography']['fontFamilies']['theme'] ) ) {
-				$settings['typography']['fontFamilies']['theme'] = array();
-			}
 		}
 
+		// Make sure there are no duplicate values with different names (array keys).
+		$settings['typography']['fontFamilies']['theme'] = array_unique( $settings['typography']['fontFamilies']['theme'], SORT_REGULAR );
```

Comment:
The `typography.fontFamilies.theme` might not be set if there is no variation or all of them don't have the font definition.

Reply:
I agree. If `$settings['typography']['fontFamilies']['theme']` is not initialliy set and there no `$variations`, this could become an issue.

Reply:
Right, good catch, thanks. Fixing.